### ISSUE TITLE
CORS issue with Raster images using importJSON

### ISSUE THREAD

**Issue by franverona** [paperjs/paper.js#754]
Hello everyone!

TL;DR:

**project.importJSON** function doesn't take care about CORS issues, so **canvas.toDataURL** function will fail when trying to import a project which contains images loaded from an external domain.

When you try to load an image from an external domain (for example: Amazon S3), you have to do two tasks:

1. Setup your domain to accept petitions from other domains. For example, in Amazon S3 could be like this:

        <CORSConfiguration>
          <CORSRule>
            <AllowedOrigin>*</AllowedOrigin>
            <AllowedMethod>GET</AllowedMethod>
          </CORSRule>
        </CORSConfiguration>

2. Create an Image object on JS, set 'crossOrigin' parameter to 'anonymous', then create Paper.js Raster by linking it to previous Image. For example:

        var imgUploaded = new Image();
        imgUploaded.src = 'http://comicdrop.s3.amazonaws.com/1392091544514_test.png';
        imgUploaded.setAttribute('crossOrigin', 'anonymous');

        imgUploaded.onload = function () {
          var r = new Raster(imgUploaded);
          project.view.update();
        };

Ok, this works when adding images but, there is a problem when trying to import a Paper.js project by using project.importJSON function. By default, this function doesn't take care about CORS issues, so images loaded from external domains will load, but when you try to do a canvas.toDataURL, you will get this error: Uncaught SecurityError: Failed to execute 'toDataURL' on 'HTMLCanvasElement': Tainted canvases may not be exported.

Regards,

Fran

**Comment by lehni** [comment]
We can't always set crossOrigin to 'anonymous' as that will create other issues for sure. But we can expose it as an attribute on Raster, along with `source` and all the other properties. This will give you the possibility to create Rasters in this way:

    var raster = new Raster({
       crossOrigin: 'anonymous',
       source: 'http://assets.paperjs.org/images/marilyn.jpg',
       position: view.center
    });

    console.log(view.element.toDataURL("image/png").substring(0, 32));

And the attribute should also be exported in JSON.
